File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/MeasureReport.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type MeasureReportGroupPopulation = BackboneElement & {
    code?: CodeableConcept;
    count?: number;
    subjectResults?: Reference;
};

export type MeasureReportGroupStratifierStratumComponent = BackboneElement & {
    code: CodeableConcept;
    value: CodeableConcept;
};

export type MeasureReportGroupStratifierStratumPopulation = BackboneElement & {
    code?: CodeableConcept;
    count?: number;
    subjectResults?: Reference;
};

export type MeasureReportGroupStratifierStratum = BackboneElement & {
    value?: CodeableConcept;
    component?: MeasureReportGroupStratifierStratumComponent[];
    population?: MeasureReportGroupStratifierStratumPopulation[];
    measureScore?: Quantity;
};

export type MeasureReportGroupStratifier = BackboneElement & {
    code?: CodeableConcept[];
    stratum?: MeasureReportGroupStratifierStratum[];
};

export type MeasureReportGroup = BackboneElement & {
    code?: CodeableConcept;
    population?: MeasureReportGroupPopulation[];
    stratifier?: MeasureReportGroupStratifier[];
    measureScore?: Quantity;
};

export type MeasureReport = DomainResource & {
    evaluatedResource?: Reference[];
    date?: string;
    group?: MeasureReportGroup[];
    type: string;
    measure: string;
    reporter?: Reference;
    status: string;
    identifier?: Identifier[];
    period: Period;
    improvementNotation?: CodeableConcept;
    subject?: Reference;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/MeasureReport.py
from typing import Optional, List
from base import Period
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class MeasureReport_Group_Population(BackboneElement):
    code: Optional[CodeableConcept] = None
    count: Optional[int] = None
    subject_results: Optional[Reference] = None

class MeasureReport_Group_Stratifier_Stratum_Component(BackboneElement):
    code: CodeableConcept
    value: CodeableConcept

class MeasureReport_Group_Stratifier_Stratum_Population(BackboneElement):
    code: Optional[CodeableConcept] = None
    count: Optional[int] = None
    subject_results: Optional[Reference] = None

class MeasureReport_Group_Stratifier_Stratum(BackboneElement):
    value: Optional[CodeableConcept] = None
    component: Optional[List[MeasureReport_Group_Stratifier_Stratum_Component]] = None
    population: Optional[List[MeasureReport_Group_Stratifier_Stratum_Population]] = None
    measure_score: Optional[Quantity] = None

class MeasureReport_Group_Stratifier(BackboneElement):
    code: Optional[List[CodeableConcept]] = None
    stratum: Optional[List[MeasureReport_Group_Stratifier_Stratum]] = None

class MeasureReport_Group(BackboneElement):
    code: Optional[CodeableConcept] = None
    population: Optional[List[MeasureReport_Group_Population]] = None
    stratifier: Optional[List[MeasureReport_Group_Stratifier]] = None
    measure_score: Optional[Quantity] = None

class MeasureReport(DomainResource):
    evaluated_resource: Optional[List[Reference]] = None
    date: Optional[str] = None
    group: Optional[List[MeasureReport_Group]] = None
    type: str
    measure: str
    reporter: Optional[Reference] = None
    status: str
    identifier: Optional[List[Identifier]] = None
    period: Period
    improvement_notation: Optional[CodeableConcept] = None
    subject: Optional[Reference] = None